### PR TITLE
implemented dom0 qubes.GetRandomizedTime

### PR DESCRIPTION
**Pull request by adrelanos**
Required for fixing 'sys-whonix doesn't connect to Tor after system suspend'.

https://github.com/QubesOS/qubes-issues/issues/1764

### CODE REVIEW

File: qubes-rpc/qubes.GetRandomizedTime [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+## Get a random 0 or 1.
+## Will use this to decide to use plus or minus.
+##
+## Thanks to
+## http://linux.byexamples.com/archives/128/generating-random-numbers/
+ZERO_OR_ONE="$(( 0+($(od -An -N2 -i /dev/random) )%(0+2) ))"
+
+## Create a random number between 0 and 180.
+DELAY="$(( $(od -An -N2 -i /dev/random)%(180-0+1) ))"
```
Comment (marmarek) [inline qubes-rpc/qubes.GetRandomizedTime]: Why use different methods of getting random number? `shuf` usage below seems much more readable. And according to the second answer in https://stackoverflow.com/questions/22887891/how-can-i-get-a-random-dev-random-number-between-0-and-999999999-in-bash, using modulo doesn't provide uniform distribution.

**Comment by marmarek** [comment]
Other than the above comment, looks good!

File: qubes-rpc/qubes.GetRandomizedTime (hunk 1 above)
Comment (adrelanos) [inline qubes-rpc/qubes.GetRandomizedTime]: No more reason to do so. (No longer excluding +/- 5.) Much more readable indeed! Fixed that.

(If it looks good, I'll also fix that in bootclockrandomization.)

File: Makefile [hunk 2]
```diff
@@ -76,8 +76,10 @@ endif
 	cp qubes-rpc-policy/qubes.NotifyUpdates.policy $(DESTDIR)/etc/qubes-rpc/policy/qubes.NotifyUpdates
 	cp qubes-rpc-policy/qubes.NotifyTools.policy $(DESTDIR)/etc/qubes-rpc/policy/qubes.NotifyTools
 	cp qubes-rpc-policy/qubes.GetImageRGBA.policy $(DESTDIR)/etc/qubes-rpc/policy/qubes.GetImageRGBA
+	cp qubes-rpc-policy/qubes.GetRandomizedTime.policy $(DESTDIR)/etc/qubes-rpc/policy/qubes.GetTime
```
Comment (marmarek) [inline Makefile]: Wrong target filename.

**Comment by marmarek** [comment]
You need to add new files to `rpm_spec/core-dom0.spec`

File: Makefile (hunk 2 above)
Comment (adrelanos) [inline Makefile]: Fixed.

**Comment by adrelanos** [comment]
Just now added. Please have a look.

**Pull request merged**